utils/io/content/contentreader.go

```diff
@@ -0,0 +1,158 @@
+package content
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"os"
+	"sync"
+
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+type ContentReader struct {
+	// Response data file path.
+	filePath, arrayKey string
+	// The objects from the response data file are being pushed to the data channel.
+	dataChannel chan map[string]interface{}
+	buffer      []map[string]interface{}
+	errorsQueue *utils.ErrorsQueue
+	once        *sync.Once
+}
+
+func NewContentReader(filePath string, arrayKey string) *ContentReader {
+	self := ContentReader{}
+	self.filePath = filePath
+	self.arrayKey = arrayKey
+	self.dataChannel = make(chan map[string]interface{}, 50000)
```

Comment:
There's probably no nedd for a large channel. I suggest replacing 50000 with 100. Let's change the size of the errorsQueue accordingly.